pydevices/MitDevices/incaa_dio2.py

```diff
@@ -0,0 +1,433 @@
+                function = _chan.function
+                if self.debug:
+                    print('function for %s is %s' % (str(chan), function))
+                if function == 'PULSE':
+                    trig_mode, trig_event = _chan.trigger
+                    cyclic = _chan.cyclic
+                    init_level_1 = _chan.init_level_1
+                    init_level_2 = _chan.init_level_2
+                    delay = _chan.delay
+                    duration = _chan.duration
+                    if self.debug:
+                        print('DIO2HWSetPulseChan(0, $1, $2, $3, $4, $5, $6, $7, $8, $9)', board_id, channel, trig_mode, cyclic,
+                                                init_level_1, init_level_2, delay, duration, trig_event)
+                    self.mds_value('DIO2HWSetPulseChan(0, $1, $2, $3, $4, $5, $6, $7, $8, $9)', board_id, channel, trig_mode, cyclic,
+                                                init_level_1, init_level_2, delay, duration, trig_event)
+                elif function == 'CLOCK':
+                    frequency = _chan.frequency_1
+                    duty_cycle = _chan.duty_cycle
+                    if self.debug:
+                        print('DIO2HWSetClockChan(0, $, $, $, $)', board_id, channel, frequency, duty_cycle)
+                    self.mds_value('DIO2HWSetClockChan(0, $, $, $, $)', board_id, channel, frequency, duty_cycle)
+                    _chan.chan_prop('clock').record = MDSplus.Range(None, None, 1.0/frequency)
+                elif function == 'GCLOCK':
+                    trig_mode, trig_event = _chan.trigger
+                    frequency = _chan.frequency_1
+                    period = long((1. / frequency) / 1E-7 + 0.5) * 1E-7
+                    delay = _chan.delay
+                    duration = _chan.duration
+                    if self.debug:
+                        print(self, 'DIO2HWSetGClockChan(0, $1, $2, $3, $4, $5, $6, $7)', board_id, channel, trig_mode, frequency,
+                                  delay, duration, trig_event)
+                    self.mds_value(self, 'DIO2HWSetGClockChan(0, $1, $2, $3, $4, $5, $6, $7)', board_id, channel, trig_mode, frequency,
+                                  delay, duration, trig_event);
+                    _chan.chan_prop('clock').record = MDSplus.Range(MDSplus.Add(chan.TRIGGER,chan.DELAY), 
+                                                                    MDSplus.Add(MDSplus.Add(chan.TRIGGER,chan.DELAY),chan.DURATION), 
+                                                                    period)
+                elif function == 'DCLOCK':
+                    trig_mode, trig_event = self.get_trigger(chan)
+                    frequency_1 = _chan.frequency_1
+                    period_1 = long((1. / frequency_1) / 1E-7 + 0.5) * 1E-7
+                    frequency_2 = _chan.frequency_2 
+                    period_2 = long((1. / frequency_1) / 1E-7 + 0.5) * 1E-7
```

Comment:
This should be (1. / frequency_2), I believe.

Reply:
fixed now